### ISSUE TITLE
Use of Java lamda disables gradle optimizations

### ISSUE THREAD

**Issue by dalewking** [icerockdev/moko-resources#450]
> Execution optimizations have been disabled for task ':xxxxxxx:packageDebugAssets' to ensure correctness due to the following reasons:
>   - Additional action of task ':xxxxxxx:packageDebugAssets' was implemented by the Java lambda 'dev.icerock.gradle.AndroidPluginLogic$setAssetsDirsRefresh$2$$Lambda$1050/0x0000000800cc6380'. Reason: Using Java lambdas is not supported as task inputs. Please refer to https://docs.gradle.org/7.5/userguide/validation_problems.html#implementation_unknown for more details about this problem.

**Comment by dalewking** [comment]
The doFirst in this code is the issue:
```
    private fun setAssetsDirsRefresh() {
        // without this code Android Gradle Plugin not copy assets to aar
        project.tasks
            .matching { it.name.startsWith("package") && it.name.endsWith("Assets") }
            .configureEach { task ->
                task.doFirst {
                    val android = project.extensions.getByType<BaseExtension>()
                    val assets = android.sourceSets.getByName("main").assets
                    assets.setSrcDirs(assets.srcDirs)
                }
            }
    }
```

Converting that to an anonymous object this should fix it or actually I'm not even sure why this is done in a configureEach block as it in no way modifies the tasks. Isn't the body of the doFirst just something that should be done once when the plugin is applied?

**Comment by Alex009** [comment]
>Isn't the body of the doFirst just something that should be done once when the plugin is applied?

comment describe this - "without this code Android Gradle Plugin not copy assets to aar".
AGP have strange logic and not use generated files if we just setup dirs at initialization

**Comment by dalewking** [comment]
According to your comment here (https://github.com/icerockdev/moko-resources/issues/353#issuecomment-1179713713) just setting it globally fixed it. But either way you should not use a Java lambda. An inner class works fine.

**Comment by Alex009** [comment]
Solution from comment require additional setup in consumer project. When gradle plugin do this - agp not read generated directories

**Comment by dalewking** [comment]
I do not see how what you are doing in that lambda in any way does anything different than setting it in the plugin. the only thing different is that you are forcing the packageXXXXXAssets task to run every time. That sounds to me like the issue is that you instead need to make the packageXXXXAssets tasks depend on the generation task so that generation will definitely by ran

**Comment by Alex009** [comment]
you can check by self how without this code android will lost assets directory. it's bug of AGP 7.4. if you can give better solution - just send pull request, it will help :)

**Comment by Alex009** [comment]
will be released in 0.21.2
